### PR TITLE
feat(sft): supervised fine-tuning on demonstrations as a recipe, with the skill stream as its example

### PR DESCRIPTION
**Pull request by MikeZheng777**
## Motivation

The self-distillation recipes of #502 need SFT on the same demonstrations as their control arm: the same reports, the same stack and optimizer, only the loss differs. Reef had no SFT recipe, only a test stub named `sft`.

## Changes

- `recipes/sft/`: the recipe, a processor that renders the report's demonstration as the assistant turn of the recorded request with the served model's chat template, the objective, and the `sft` Slime family on the stock `sft_loss` (advantages refused rather than dropped).
- `reef.core.reports.TeacherContextReport`: the shared report of the demonstration recipes (a rollout's receipt and the privileged text as `context`); `reef.core.chat_request` holds the recorded-request readers the processors share.
- `recipes/sft/examples/skill_stream/`: Tool Use, then Science Q&A (the SFT arm of Figure 3 of arXiv:2601.19897) through reef-eval, each stage a Harbor task whose judge scores the served model on both skills; recorded run and curves under `results/skill_stream/`.
- Two fixes on the colocated full-weight publication path, which no bundled deployment exercised: the HF export files Slime's global-named actor backup under the Bridge's single pipeline stage, and the colocated engine send uses Slime's `weight_version` keyword.
- `tests/conftest.py` registers only the `pg` stub; `sft` is the cookbook's.

## Compatibility and operational impact

New recipe, report type and core module; no change to existing recipes or wire contracts. The two adapter fixes only touch a path that raised before (`KeyError` in the export, `TypeError` in the send); stage-named backups and the LoRA transport are untouched.

## Verification

- `pytest` on the host: the sft pipeline, report, contract, config and skill-stream suites pass; in the training container (`reef-ceobench:latest`): `tests/reef_service/test_slime_bridge.py test_sao_configs.py tests/slime_backend ...`, 583 passed.
- `pre-commit run --files ...` and `python -m mypy`: clean.
- The example ran on 4 H100s (colocated, TP4): Tool Use 41.2% → 68.0% over 252 steps, Science Q&A 26.0% → 69.6% by step 100 of its stage while Tool Use fell to 48.5%; the Science stage was stopped at step 108/167 to free the GPUs, as the README says.

## AI assistance

Claude Code drafted the implementation, tests and documentation under my direction; I reviewed every change and ran the experiment.

## Checklist

- [x] The change is focused and contains no unrelated cleanup.
- [x] Tests cover behavior changes, or this pull request does not change behavior.
- [x] Public interface changes include contract tests, or no public interface changes are present.
- [x] Affected user and developer documentation is updated, or no documentation update is required.
- [x] Root README changes update both languages and `README.i18n.yaml`, or do not affect README content.
- [x] An accepted RFC is linked, or this change does not require an RFC.
- [x] Relevant pre-commit and test checks pass.
- [x] Non-trivial AI assistance is disclosed above, or none was used.

---
<a href="https://app.blacksmith.sh/Human-Agent-Society/codesmith/reef/pr/531"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1792271785&installation_model_id=437034&pr_number=531&repository=Human-Agent-Society%2Freef&return_to=https%3A%2F%2Fgithub.com%2FHuman-Agent-Society%2Freef%2Fpull%2F531&signature=ea79f842a7b12cc322bc2cf014d73d3a4d5787d8b51b3b12eea83f4b736bae6b"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by MikeZheng777** [comment]
Closing: the SFT recipe is the control arm, not a recipe on its own; it lands with the SDFT branch (hanzheng/sdft-distill), which carries these commits.
